.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,206 @@
+
+# Considerations when writing sniffs
+
+## Public properties
+
+When writing sniffs, always remember that any `public` sniff property can be overruled via a custom ruleset by the end-user.
+Only make a property `public` if that is the intended behaviour.
+
+When you introduce new `public` sniff properties, or your sniff extends a class from which you inherit a `public` property, please don't forget to update the [public properties wiki page](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Customizable-sniff-properties) with the relevant details once your PR has been merged into the `develop` branch.
+
+# Unit Testing
+
+## Pre-requisites
+
+* PHP 5.4 or higher
+* WordPress-Coding-Standards 1.0.0 or higher
+* PHP_CodeSniffer 3.3.0 or higher
+* PHPUnit 4.x, 5.x, 6.x or 7.x
+* PHPCompatibility 8.0 or higher
+
+### Composer setup
+
+Use Composer to install all the necessary dependencies to write and run unit tests. Running
+
+```sh
+composer install
+```
+
+will install PHP_CodeSniffer, WordPress Coding Standards, PHPUnit, PHPCompatibility and security advisories which ensures that your application doesn't have installed dependencies with known security vulnerabilities.
+
+### Other setups
+
+If you have PHP_CodeSniffer (PHPCS) and/or WordPress Coding Standards (WordPressCS) already installed, for instance as git clones, because you either contribute to these libraries as well or you want to develop for or test WPThemeReview with bleeding edge versions of either PHP_CodeSniffer or WordPress Coding Standards, you need to take some additional steps to make it all work.
+
+First, make sure you also have PHPCompatibility and make sure the `installed_paths` for PHP_CodeSniffer is setup correctly.
+
+#### Method 1
+
+Set up environment variables to point to PHP_CodeSniffer and WordPress Coding Standards:
+
+```sh
+PHPCS_DIR: I:/000_GitHub/PHPCS/PHP_CodeSniffer/
+WPCS_DIR: I:/000_GitHub/PHPCS/WordPressCS/
+```
+
+If you do that, everything should work as expected.
+
+#### Method 2
+
+If you do not want to set up environment variables on your system
+
+* Add a file called `.pathtowpcs` to the project root. The only content in that file should be the full absolute path to your WordPressCS install.
+* Copy the `phpunit.xml.dist` file, rename the copied file `phpunit.xml` and make sure it's in the project root.
+Now add the following to that file, adjusting the paths to reflect those on your system:
+
+```xml
+  <php>
+    <env name="PHPCS_DIR" value="path/to/PHP_CodeSniffer/"/>
+    <env name="WPCS_DIR" value="path/to/WordPressCS/"/>
+  </php>
+```
+
+(and don't remove the existing line within the `<php>` block.)
+
+## Writing and running unit tests
+
+The most iportant thing when writing sniffs intended for the theme review, is to have an ample of examples. This makes writing sniffs a lot easier, because you can test agains the give nexamples.
```

Comment:
`most iportant thing` => `most important thing`

`an ample of examples` => `ample example code`

`give nexamples` => `given examples`